extensions/cloud/portability-cloud-google/src/main/java/org/datatransferproject/cloud/google/GoogleDtpInternalMetricRecorder.java

```diff
@@ -15,28 +15,211 @@
+      TagMetadata.create(TagMetadata.TagTtl.UNLIMITED_PROPAGATION);
+
+  private static final Measure.MeasureLong JOB_STARTED = Measure.MeasureLong.create(
+      "job_start",
+      "Number of jobs that were started",
+      "count");
+
+  private static final Measure.MeasureLong JOB_FINISHED = Measure.MeasureLong.create(
+      "job_finished",
+      "Num er of jobs that finished",
+      "count");
+
+  private static final Measure.MeasureLong JOB_FINISHED_DURATION = Measure.MeasureLong.create(
+      "job_finished_duration",
+      "Duration of a job",
+      "ms");
+
+  private static final Measure.MeasureLong EXPORT_PAGE_ATTEMPT = Measure.MeasureLong.create(
+      "export_page_attempt",
+      "An single export attempt",
+      "count");
+
+  private static final Measure.MeasureDouble EXPORT_PAGE_ATTEMPT_DURATION =
+      Measure.MeasureDouble.create(
+          "export_page_attempt_duration",
+          "Duration of a single export attempt in MS",
+          "ms");
+
+  private static final Measure.MeasureLong EXPORT_PAGE = Measure.MeasureLong.create(
+      "export_page",
+      "An export attempt including all retries",
+      "count");
+
+  private static final Measure.MeasureDouble EXPORT_PAGE_DURATION =
+      Measure.MeasureDouble.create(
+          "export_page_duration",
+          "Duration of am export page including retries in MS",
+          "ms");
+
+  private static final Measure.MeasureLong IMPORT_PAGE_ATTEMPT = Measure.MeasureLong.create(
+      "import_page_attempt",
+      "An single import attempt",
+      "count");
+
+  private static final Measure.MeasureDouble IMPORT_PAGE_ATTEMPT_DURATION =
+      Measure.MeasureDouble.create(
+          "import_page_attempt_duration",
+          "Duration of a single import attempt in MS",
+          "ms");
+
+  private static final Measure.MeasureLong IMPORT_PAGE = Measure.MeasureLong.create(
+      "import_page",
+      "An import attempt including all retries",
+      "count");
+
+  private static final Measure.MeasureDouble IMPORT_PAGE_DURATION =
+      Measure.MeasureDouble.create(
+          "import_page_duration",
+          "Duration of am imoprt page including retries in MS",
+          "ms");
+
+  private static final Measure.MeasureLong GENERIC_COUNT = Measure.MeasureLong.create(
+      "generic_count",
+      "A generic counter that services can use to hold arbitrary metrics",
+      "count");
+
+  private static final Measure.MeasureLong GENERIC_DURATION = Measure.MeasureLong.create(
+      "generic_duration",
+      "A generic counter that services can use to hold arbitrary duration metrics",
+      "ms");
+
+  private static final Measure.MeasureLong GENERIC_BOOLEAN = Measure.MeasureLong.create(
+      "generic_boolean",
+      "A generic counter that services can use to hold arbitrary boolean metrics",
+      "count");
+
+  private final ViewManager viewManager;
+
+  // This is needed because Stackdriver can only be initialized once and the
```

Comment:
Ok, we should work through the Google cloud packages initialization story at some point but this is very cool for now.